File: 백준/Silver/9012. 괄호/괄호.py
T = int(input())

for t in range(T):
    f = True
    vps = []
    ps = input()

    for i in ps:
        if i == '(':
            vps.append(i)
        elif '(' in vps and i == ')':
            vps.pop()
        else:
            break

    else:
        if len(vps) == 0:
            f = False     

    print('YES' if f == False else 'NO')

File: 백준/Silver/9012. 괄호/괄호.swift
import Foundation

let n = Int(readLine()!)!
for i in 0..<n {
    let input = readLine()!
    var count = 0
    var answer = ""
    for ele in input {
        if ele == "(" {
            count += 1
        } else if ele == ")" {
            count -= 1
        }

        if count < 0 {
            answer = "NO"
            break
        }
    }
    if !answer.isEmpty {
        print(answer)
    } else {
        answer = count == 0 ? "YES" : "NO"
        print(answer)
    }
}